### PR TITLE
Allows for a global fallback configuration used for all bound actors

### PR DESCRIPTION
**Pull request by autodidaddict**
Signed-off-by: Kevin Hoffman <autodidaddict@users.noreply.github.com>

### CODE REVIEW

**Review by stevelr — APPROVED** [review]
If the linkdef default config is the same as provider config, consider this approved. I can implement a model for that separation in example code and other providers.

File: lattice-controller/src/main.rs [hunk 1]
```diff
@@ -184,7 +215,12 @@ fn get_actor(ctx: &Context) -> RpcResult<String> {
 }
 
 async fn get_config(lc: &LatticeControllerProvider, actor_id: &str) -> ConnectionConfig {
-    lc.configs.read().await.get(actor_id).unwrap().clone()
+    lc.configs
+        .read()
+        .await
+        .get(actor_id)
+        .unwrap_or(&lc.fallback_config)
```
[the remark this answers is not in the document]
Comment (autodidaddict) [reply lattice-controller/src/main.rs:222]: Yeah I think how each provider handles the global `config_json` field from the `HostData` struct will be different. I suspect many of them will simply ignore it (e.g. a database client provider). In this case, the way the logic works is that you can choose to provide a single NATS connection config for all bound actors, or you can override that for some actors, or you can choose to not supply the global configuration at all.

**Comment by autodidaddict** [comment]
> If the linkdef default config is the same as provider config, consider this approved. I can implement a model for that separation in example code and other providers.

Yes. If the host data configuration is supplied, that will be considered the "fallback" config for when no linkdef configuration is present. If that is not supplied, then the `Default::default()` value will be used, which is the localhost/anonymous (e.g. leaf-node friendly) config. If a specific configuration is supplied with the link definition for an actor, then that one will be used instead of the default/fallback.

**Comment by autodidaddict** [comment]
From the actor perspective, if the provider was started with a configuration JSON (which is an option on the start provider command), then you can provide an empty values hashmap for all of the actors and they'll all use the singleton NATS configuration for lattice manipulation. The "old way" still works fine, so long as you provide a `config_json` or a `config_b64` as the linkdef value.

**Review by brooksmtownsend — APPROVED** [review]
So, the net result of this PR is that the lattice controller provider is configured such that any linked actors without specified values automatically connect to the lattice that the provider is running in?

I had to poke around a bit since initially this felt like it might be a vulnerability (if we throw NATS auth out the window) e.g. if you publish a link with a malicious actor the provider and it can automatically issue lattice control commands. IMO at this point if you are able to start a malicious actor, publish link definitions, and you already have the lattice connection information, it's MOOT. LGTM

**Comment by autodidaddict** [comment]
Even then said malicious actor would only be able to use the previously configured fallback connection, not an arbitrary one.

**Pull request merged**